pkg/controller/cluster/cluster_controller.go

```diff
@@ -25,8 +25,14 @@ import (
 	"fmt"
 	"net/http"
 	"reflect"
+	"strings"
 	"time"
 
+	iamv1alpha2 "kubesphere.io/api/iam/v1alpha2"
+
+	kubesphere "kubesphere.io/kubesphere/pkg/client/clientset/versioned"
+	iamv1alpha2listers "kubesphere.io/kubesphere/pkg/client/listers/iam/v1alpha2"
```

Comment:
Need to be grouped correctly, this is getting more and more chaos

Reply:
> Need to be grouped correctly, this is getting more and more chaos

It's sorted by `goimports`, using `controller-runtime` instead of the worker queue and informers can keep the code cleanly.

Reply:
> It's sorted by goimports

`goimports` will sort packages within a group, but if they are not originally in a group, `goimports` will do nothing, try this:

```go
import (
	"k8s.io/apimachinery/pkg/api/errors"

	"k8s.io/client-go/discovery"

	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"k8s.io/client-go/util/retry"

	"github.com/emicklei/go-restful"

	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"

	"k8s.io/apimachinery/pkg/runtime/schema"

	urlruntime "k8s.io/apimachinery/pkg/util/runtime"

	"k8s.io/apimachinery/pkg/util/sets"

	"k8s.io/apiserver/pkg/endpoints/handlers/responsewriters"
	unionauth "k8s.io/apiserver/pkg/authentication/request/union"
)
```

> using controller-runtime instead of the worker queue and informers can keep the code cleanly.

You misunderstood, I'm just saying the group is getting more and more.

Reply:
got it